### DISCUSSION TITLE
Dynamic padding for neovim

### DISCUSSION THREAD

**Discussion by erlestor** [wezterm/wezterm#6497] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### Which Wayland compositor or X11 Window manager(s) are you using?

komorebi

### WezTerm version

20240203-110809-5046fc22

## Actual question 

Is there a way to set the padding to 0 only when I have neovim open in the active tab?

## Background

I often have neovim in one tab and a development server running in another. When Im in the neovim tab I want 0 padding, but when i go the powershell tab with my dev server running I want 8px padding.

I've seen some gh issues mentioning the same, but the best solution I found was to use the "VimEnter" and "VimLeavePre" events to set change padding. The issue is that I often leave neovim running while i open another tab. Since I never left vim my padding is still set to zero. 

I've also tried using `pane:get_foreground_process_name()` to check if neovim is currently open in the active tab, but this usually gives me the path to  eg. "node.exe" which would be the same as when running a node command in powershell

**Comment by nekowinston** [comment]
You could try the approach I posted in https://github.com/wez/wezterm/discussions/3839#discussioncomment-6123338

**Comment by erlestor** [comment, marked as answer]
Thanks a lot! I changed it a bit so the padding is 0 as long as one of the panes in the active tab has an alt screen active

```lua
wezterm.on("update-status", function(window, _)
    local tab = window:active_tab()
    local panes = tab:panes()
    local alt_screen_active = false
    
    for i = 1, #panes, 1 do
        local pane = panes[i]
        if pane:is_alt_screen_active() then
	    alt_screen_active = true
	    break
        end
    end
    
    if alt_screen_active then
    window:set_config_overrides({
        window_padding = { left = 0, right = 0, top = 0, bottom = 0 },
    })
    else
    window:set_config_overrides({
        window_padding = default_padding,
    })
    end
end)
```
